pkg/gce-pd-csi-driver/controller.go

```diff
@@ -332,10 +325,7 @@ func (gceCS *GCEControllerServer) createVolumeInternal(ctx context.Context, req
 	// Apply Parameters (case-insensitive). We leave validation of
 	// the values to the cloud provider.
 	params, err := gceCS.parameterProcessor().ExtractAndDefaultParameters(req.GetParameters(), gceCS.Driver.extraVolumeLabels, gceCS.Driver.extraTags)
-	diskTypeForMetric = params.DiskType
-	enableConfidentialCompute = strconv.FormatBool(params.EnableConfidentialCompute)
-	hasStoragePools := len(params.StoragePools) > 0
-	enableStoragePools = strconv.FormatBool(hasStoragePools)
+	metrics.UpdateRequestMetadataFromParams(ctx, params)
```

Comment:
Why can't we put this as the first line in createVolumeInternal? 

We are returning InvalidArgument errors in several cases before we call this. I think that means this won't  be reported to the metric, correct? We don't really count those InvalidArgument requests to calculate any success rates, but this will make the PDCSI controller stackdriver logs, not match what the metric actually reports which sometimes makes it a little more confusing on debugging the alerts.

Reply:
SG, moved up the parameter extraction.